File: 2 sum/solution_bruteforce.py
from typing import List


def two_sum_bruteforce(nums: List[int], target: int) -> List[int]:
    """Brute-force search over all index pairs."""

    n = len(nums)

    for i in range(n):
        for j in range(i + 1, n):
            if nums[i] + nums[j] == target:
                return [i, j]

    return [-1, -1]

File: 2 sum/solution_bruteforce.js
/**
 * Brute-force search evaluates every pair of indices.
 * @param {number[]} nums
 * @param {number} target
 * @returns {[number, number]}
 */
function twoSumBruteforce(nums, target) {
  for (let i = 0; i < nums.length; i += 1) {
    for (let j = i + 1; j < nums.length; j += 1) {
      if (nums[i] + nums[j] === target) {
        return [i, j];
      }
    }
  }

  return [-1, -1];
}

module.exports = { twoSumBruteforce };
